lib/src/test/java/tech/relaycorp/relaydroid/messaging/SendMessageTest.kt

```diff
@@ -0,0 +1,84 @@
+package tech.relaycorp.relaydroid.messaging
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.check
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert.assertArrayEquals
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import tech.relaycorp.poweb.PoWebClient
+import tech.relaycorp.relaydroid.PublicThirdPartyEndpoint
+import tech.relaycorp.relaydroid.test.FirstPartyEndpointFactory
+import tech.relaycorp.relaydroid.test.MessageFactory
+import tech.relaycorp.relaynet.bindings.pdc.ServerConnectionException
+import tech.relaycorp.relaynet.messages.Parcel
+import java.time.Duration
+
+internal class SendMessageTest {
+
+    private val poWebClient = mock<PoWebClient>()
+    private val coroutineScope = TestCoroutineScope()
+    private val subject = SendMessage({ poWebClient }, coroutineScope.coroutineContext)
+
+    @Test(expected = InvalidMessageException::class)
+    fun invalidMessage() = coroutineScope.runBlockingTest {
+        val message = OutgoingMessage(
+            ByteArray(0),
+            FirstPartyEndpointFactory.build(),
+            PublicThirdPartyEndpoint("http://example.org")
+        )
+
+        subject.send(message)
+    }
+
+    @Test
+    fun deliverParcelToPublicEndpoint() = coroutineScope.runBlockingTest {
+        val message = MessageFactory.buildOutgoing()
```

Comment:
I think it's too implicit that this is generating a message bound for a public endpoint and we could change it inadvertently.

One way to address this is by requiring Relaynet's `RecipientAddressType` enum as an argument.